pkg/controller/control.go

```diff
@@ -768,23 +768,34 @@ func (c *Controller) WriteAt(b []byte, off int64) (int, error) {
 }
 
 func (c *Controller) writeInWOMode(b []byte, off int64) (int, error) {
-	bufLen := int64(len(b))
+	bufLen := len(b)
 	// buffer b is defaultSectorSize aligned
-	if (bufLen >= defaultSectorSize) && (off%defaultSectorSize == 0) && ((off+bufLen)%defaultSectorSize == 0) {
+	if (bufLen >= defaultSectorSize) && (off%defaultSectorSize == 0) && ((off+int64(bufLen))%defaultSectorSize == 0) {
 		return c.backend.WriteAt(b, off)
 	}
```

Comment:
NIT: This can be simplified, since if bufLen == 0, we can either let it pass down or exit early here. Since I am not sure that there is any reason in continuing with the below operations in the case where the write has a length of 0?

Explanation:
off%defaultSectorSize == 0 --> implies that offset is a multiple of sector size
bufLen % defaultSectorSize == 0 --> implies that len is a multiple of sector size
therefore the sum of (len + offset) must also be a multiple of sector size.

So this can be simplified to:
(off%defaultSectorSize == 0) && (bufLen%defaultSectorSize == 0)

Reply:
The simplification isn't as important, but I am concerned about the buflen == 0 case which would trigger the rmw cycle instead of the direct write. Since a quick check didn't show an exit condition.

Reply:
> pre-4k-post case?

Yeah, we have it 
```
	// Test case #3
	buf = makeByteSliceWithInitialData(4096+1024, newVal)
	off = 4096 - 512
```

> So this can be simplified to: (off%defaultSectorSize == 0) && (bufLen%defaultSectorSize == 0)

Thank you. I updated

> Test case len(buf) == 0 and off%defaulSectorSize==0 is added

```
	// Test case #8
	buf = makeByteSliceWithInitialData(0, newVal)
	off = 4096 * 3
```